tier1/keyvalues3.cpp

```diff
@@ -1237,7 +1294,7 @@ void KeyValues3::CopyFrom( const KeyValues3* pSrc )
 			{
 				case KV3_TYPEEX_ARRAY:
 				{
-					PrepareForType( KV3_TYPEEX_ARRAY, KV3_SUBTYPE_ARRAY );
+					SetToEmptyArray();
 					m_Data.m_pArray->CopyFrom( this, pSrc->m_Data.m_pArray );
```

Comment:
I guess you'd want to use GetArray() in here instead to match the rest of the changes.

Reply:
It's already kv3(array/table) since the switch check

Reply:
I'm talking about changing m_Data.m_pArray to GetArray() in here, since that's what you did with table few lines below